lucille-core/src/test/java/com/kmwllc/lucille/core/RunnerTest.java

```diff
@@ -856,4 +856,18 @@ public void testStringifyValidationNoExceptions() {
     Map<String, List<Exception>> exceptions = new LinkedHashMap<>();
     assertEquals("Pipeline Configuration is valid.", Runner.stringifyValidation(exceptions, "Pipeline"));
   }
+
+  /**
+   * Attempt to reproduce a deadlock that was seen in a pipeline that dropped a high percentage of docs, but not all docs.
+   * The deadlock is also sensitive to how many docs are processed in total, the batch size, and the logging configuration.
+   *
+   * This test will possibly hang if the deadlock potential still exists in the code. It should pass in several seconds
+   * assuming the deadlock has been fixed, otherwise it will time out and fail after 30 seconds.
+   */
+  @Test(timeout = 30000)
+  public void testDropDeadlock() throws Exception {
+    RunResult result =
+        Runner.run(ConfigFactory.load("RunnerTest/dropDeadlock.conf"), Runner.RunType.TEST);
```

Comment:
if we switch from Runner.RunType.TEST to Runner.RunType.LOCAL can we reduce the number of docs required to consistently reproduce this issue any further?

the main difference between RunType.TEST and RunType.LOCAL is that TEST saves all the messages that are sent during the run for later inspection. We don't need those saved messages here because we're not looking at them. It's not clear that switching to LOCAL will change the scenario meaningfully, just wondering if it happens to make a difference...